dxp/resource_injection.md

```diff
@@ -0,0 +1,106 @@
+# DXP Support for Injection of Dynamic Resources
+
+This document describes different APIs that can be used with DXP to inject resources (`.js`, `.css`, ...) to a HTML page, thus letting the user extend currently existing artifacts without any need to overwrite them (like it was previously done with hooks in older Liferay versions).
+
+# `DynamicInclude` in `portal-kernel`
+
+[`DynamicInclude`](https://github.com/liferay/liferay-portal/blob/973e6199844436c22d237e2daf4821d0f9af5362/portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/DynamicInclude.java) is the lowest level API that lets programmers inject arbitrary content in any rendered page through access to its [`HttpServletResponse`](https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html).
```

Comment:
"lowest level" → "lowest-level" (without the hyphen, it sounds like this is the lowest of the "level APIs").

Reply:
:+1: